domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -97,15 +374,69 @@ class ProfileManagementController @Inject constructor(
     return ProfileId.newBuilder().setInternalId(currentProfileId).build()
   }
 
-  /**
-   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
-   * Checks to ensure that profileId corresponds to an existing profile.
-   *
-   * @param profileId the ID corresponding to the profile being set.
-   * @return a [LiveData] that indicates the success/failure of this set operation.
-   */
-  fun setCurrentProfileId(profileId: ProfileId): LiveData<AsyncResult<Any?>> {
-    currentProfileId = profileId.internalId
-    return MutableLiveData(AsyncResult.success<Any?>(null))
+  private fun isUniqueName(newName: String, profileDatabase: ProfileDatabase): Boolean {
+    val lowerCaseNewName = newName.toLowerCase()
+    profileDatabase.profilesMap.values.forEach {
+      if (it.name.toLowerCase(Locale.getDefault()) == lowerCaseNewName) {
+        return false
+      }
+    }
+    return true
+  }
+
+  private fun saveImageToInternalStorage(avatarImagePath: Uri, profileDir: File): String {
+    val bitmap = MediaStore.Images.Media.getBitmap(context.contentResolver, avatarImagePath)
+    val imageFile = File(profileDir, "profile.png")
+    var fos: FileOutputStream? = null
```

Comment:
We can do this a bit more simply using ``use`` (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html):

```kotlin
private fun saveImageToInternalStorage(avatarImagePath: Uri, profileDir: File): String? {
  val bitmap = ...
  val imageFile = ...
  try {
    FileOutputStream(imageFile).use { fos ->
      rotateAndCompressBitmap(...)
    }
    return imageFile.absolutePath
  } catch (e: Exception) {
    return null
  }
}
```

Reply:
NB: I also updated this to return null rather than using a sentinel value for failure--null can be used as 'absent' or 'failed'.